src/components/datepicker/datepicker.ts

```diff
@@ -116,73 +122,203 @@ export class MDatepicker extends ModulVue {
+            return true;
+
+        } else if (inputValue.length === 10 && this.validateDateFormat(inputValue)) {
+            let newDate: ModulDate = new ModulDate(inputValue);
+            if (newDate.isBetween(this.minModulDate, this.maxModulDate)) {
                 this.internalCalandarErrorMessage = '';
+                return true;
             } else {
                 this.internalCalandarErrorMessage = this.$i18n.translate('m-datepicker:out-of-range-error');
+                return false;
             }
         } else {
             this.internalCalandarErrorMessage = this.$i18n.translate('m-datepicker:format-error');
+            return false;
+        }
+
+        return true;
+    }
+
+
+    // Model management
+
+    // override from InputManagement
+    @Watch('value', { immediate: true })
+    private onValueChange(value: DatePickerSupportedTypes): void {
+
+        if (this.internalDateModel !== this.convertModelToString(value)) {
+            this.internalDateModel = this.convertModelToString(value);
+            this.inputModel = this.internalDateModel;
+            this.showErrorMessage(this.inputModel);
         }
+
     }
 
-    private selectDate(selectedDate: DatePickerSupportedTypes): void {
-        this.internalCalandarErrorMessage = '';
-        this.$emit('change', this.createModelUpdateValue(selectedDate));
-        this.open = false;
+    // override from InputManagement
+    private set model(value: string) {
+        this.internalDateModel = value;
+        this.emitChange();
     }
 
-    private createModelUpdateValue(newValue: DatePickerSupportedTypes): DatePickerSupportedTypes {
-        if (newValue && newValue instanceof Date) {
-            return new Date(newValue);
-        } else {
-            return newValue;
+    private get model(): string {
+        return this.internalDateModel;
+    }
+
+    // override from InputManagement
+    public get hasValue(): boolean {
+        return !!(this.inputModel || '').toString().trim();
+    }
+
+    public emitChange(): void {
+        this.$emit('change', this.convertStringToModel(this.internalDateModel));
+    }
+
+
+    // Focus management.
+
+    // override from InputManagement
+    private onFocus(event: FocusEvent): void {
+        if (!this.open) { // open on focus
+            this.open = true;
+        }
+
+        this.as<InputManagement>().internalIsFocus = this.as<InputStateMixin>().active;
+        if (this.as<InputManagement>().internalIsFocus) {
+            this.$emit('focus', event);
+        }
+    }
+
+    // override from InputManagement
+    private onClick(event: MouseEvent): void {
+        this.as<InputManagement>().internalIsFocus = this.as<InputStateMixin>().active;
+        let inputEl: HTMLElement | undefined = this.as<InputStateMixin>().getInput();
+        if (this.as<InputManagement>().internalIsFocus && inputEl) {
+            inputEl.focus();
         }
+        this.$emit('click');
```

Comment:
Juste pour curiosité, quel est la différence entre utiliser this.$emit('click') contre utiliser @Emit('click') ?

Reply:
Utiliser le @Emit('click')

Reply:
@tharle  c'est que dans modUL lorsqu'on genere la documentation pour le site web. le script ne tient compte des @emit